src/search/common_transformer.h

```diff
@@ -105,6 +105,46 @@ struct TreeTransformer {
 
     return result;
   }
+
+  template <typename T = double>
+  static StatusOr<std::vector<T>> Binary2Vector(std::string_view str) {
```

Comment:
Hmm I tried some simple queries on Redis Stack, e.g.
```
FT.SEARCH documents "*=>[KNN 10 @doc_embedding $BLOB]" PARAMS 2 BLOB "\x12\xa9\xf5\x6c" SORTBY __vector_score DIALECT 4
```

Seems here the parameter `"\x12\xa9\xf5\x6c"` is a redis-cli level string literal and it will be converted to binary format  (i.e. a 4-byte binary string) before it is sent to the redis server.

So maybe in the server side we can just do some bit-level casts.

Reply:
Thanks for helping test it; will make the changes!
One thing is `std::bit_cast<T>` is supported until c++20, I may have to use `memcpy` instead